package/MDAnalysis/coordinates/chain.py

```diff
@@ -79,68 +79,69 @@ class ChainReader(base.ProtoReader):
+            (i.e. they all must belong to the same topology). The trajectory
+            format is deduced from the extension of each file name.
+
+            Extension: `filenames` are either a single file name or list of file
+            names in either plain file names format or ``(filename, format)``
+            tuple combination. This allows explicit setting of the format for
+            each individual trajectory file.
         skip : int (optional)
-           skip step (also passed on to the individual trajectory readers);
-           must be same for all trajectories
+            skip step (also passed on to the individual trajectory readers);
+            must be same for all trajectories
         dt : float (optional)
-          Passed to individual trajectory readers to enforce a common time
-          difference between frames, in MDAnalysis time units. If not set, each
-          reader's `dt` will be used (either inferred from the trajectory
-          files, or set to the reader's default) when reporting frame times;
-          note that this might lead an inconsistent time difference between
-          frames.
+            Passed to individual trajectory readers to enforce a common time
+            difference between frames, in MDAnalysis time units. If not set, each
+            reader's `dt` will be used (either inferred from the trajectory
+            files, or set to the reader's default) when reporting frame times;
+            note that this might lead an inconsistent time difference between
+            frames.
+        continuous : bool (optional)
+            treat all trajectories as one single long trajectory. Adds several
+            checks. Might not work with all trajectory formats! This should
+            ensure that there is no need to concatenate trajectories of GROMACS
+            simulations
         **kwargs : dict (optional)
-          all other keyword arguments are passed on to each trajectory reader
-          unchanged
-
-
-        .. versionchanged:: 0.8
-           The ``delta`` keyword was added.
-        .. versionchanged:: 0.13
-           The ``delta`` keyword was deprecated in favor of using ``dt``.
+            all other keyword arguments are passed on to each trajectory reader
+            unchanged
 
         """
         super(ChainReader, self).__init__()
 
-        self.filenames = asiterable(filenames)
-        self.readers = [core.reader(filename, **kwargs)
-                        for filename in self.filenames]
+        filenames = asiterable(filenames)
+        # Override here because single frame readers handle this argument as a
+        # kwarg to a timestep which behaves differently if dt is present or not.
+        if dt is not None:
+            kwargs['dt'] = dt
+        self.readers = np.array([core.reader(filename, **kwargs)
+                                 for filename in filenames], dtype=object)
```

Comment:
This line has an interesting bug! I convert this to a numpy array to make resorting it easier later on when the trajectory is continuous. But there is a strange effect for a PDBReader when I convert this it gets converted into an array. Why would this happen I don't see any obvious reason? Does numpy pickup on some iterator protocol?

```python
In [2]: from MDAnalysisTests.datafiles import PDB

In [3]: filenames = [PDB, PDB]

In [4]: [mda.coordinates.core.reader(fn) for fn in filenames]
/home/max/code/molecular-dynamics/mdanalysis/mdanalysis/package/MDAnalysis/topology/guessers.py:56: UserWarning: Failed to guess the mass for the following atom types: M
  "".format(', '.join(misses)))
Out[4]: 
[<PDBReader /home/max/code/molecular-dynamics/mdanalysis/mdanalysis/testsuite/MDAnalysisTests/data/adk_oplsaa.pdb with 1 frames of 47681 atoms>,
 <PDBReader /home/max/code/molecular-dynamics/mdanalysis/mdanalysis/testsuite/MDAnalysisTests/data/adk_oplsaa.pdb with 1 frames of 47681 atoms>]

In [5]: np.array([mda.coordinates.core.reader(fn) for fn in filenames], dtype=object)
/home/max/code/molecular-dynamics/mdanalysis/mdanalysis/package/MDAnalysis/topology/guessers.py:56: UserWarning: Failed to guess the mass for the following atom types: M
  "".format(', '.join(misses)))
Out[5]: 
array([[[[52.016998291015625, 43.560001373291016, 31.55500030517578],
         [51.1879997253418, 44.11199951171875, 31.722000122070312],
         [51.55099868774414, 42.827999114990234, 31.038999557495117],
         ..., 
         [105.34200286865234, 74.0719985961914, 40.987998962402344],
         [57.683998107910156, 35.32400131225586, 14.803999900817871],
         [62.96099853515625, 47.23899841308594, 3.753000020980835]]],


       [[[52.016998291015625, 43.560001373291016, 31.55500030517578],
         [51.1879997253418, 44.11199951171875, 31.722000122070312],
         [51.55099868774414, 42.827999114990234, 31.038999557495117],
         ..., 
         [105.34200286865234, 74.0719985961914, 40.987998962402344],
         [57.683998107910156, 35.32400131225586, 14.803999900817871],
         [62.96099853515625, 47.23899841308594, 3.753000020980835]]]], dtype=object)

```

Reply:
Does the reader happen to have an `__iter__()` method?